lib/terraform/basetest.pm

```diff
@@ -0,0 +1,138 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Base class for terraform tests
+#
+# Maintainer: Jose Lausuch <jalausuch@suse.de>
+
+package terraform::basetest;
+use Mojo::Base 'opensusebasetest';
+use testapi;
+use terraform::libvirt_domain;
+use terraform::host;
+use Data::Dumper;
+use Mojo::JSON 'decode_json';
+use MIME::Base64;
+
+use constant TERRAFORM_TIMEOUT => 600;
+use constant WORKDIR           => '/root/terraform';
+
+=head2 _run_terraform_cmd
+
+    _run_terraform_cmd();
+
+Runs terraform command given by C<cmd>
+
+=cut
+sub _run_terraform_cmd {
+    my ($self, $cmd) = @_;
+    my $output = script_output($cmd, TERRAFORM_TIMEOUT);
```

Comment:
what about flip the order ? in case of failure you will not get command displayed